### PR TITLE
(plugin) database::mongodb - add shards option

### PR DESCRIPTION
**Pull request by garnier-quentin**

### CODE REVIEW

File: src/database/mongodb/mode/collectionstatistics.pm [hunk 1]
```diff
@@ -151,27 +224,17 @@ Check collections statistics per databases
...
+Filter shards by name (Can use regexp).
 
-=item B<--critical-subinstance-collection-*-count>
+=item B<--warning-*> B<--critical-*>
 
-Critical threshold.
-Can be: 'documents', 'indexes'.
+Thresholds.
+Can be: 'storage-size', 'index-size', 'documents', 'indexes',
+'shard-storage-size', 'shard-index-size', 'shard-documents', 'shard-indexes'.
```
Comment (lucie-dubrunfaut) [inline src/database/mongodb/mode/collectionstatistics.pm:238]: Don't 'add-shards' option should be explain in help section ?

File: src/database/mongodb/mode/collectionstatistics.pm [hunk 2]
```diff
@@ -115,21 +168,41 @@ sub manage_selection {
...
+                        shardName => $shard_name,
+                        storageSize => $cl_stats->{shards}->{$shard_name}->{storageSize},
+                        totalIndexSize => $cl_stats->{shards}->{$shard_name}->{totalIndexSize},
+                        count => $cl_stats->{shards}->{$shard_name}->{count},
+                        nindexes => $cl_stats->{shards}->{$shard_name}->{nindexes}
+                    };
+                }
+            }
         }
     }
```
Comment (lucie-dubrunfaut) [inline src/database/mongodb/mode/collectionstatistics.pm:208]: Catch error if filter_shard option return empty result should be add here 
Something like this :

```
    if (scalar(keys %{$self->{databases}->{$database}->{shards}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => 'No shard found with this filter');
        $self->{output}->option_exit();
    }
```

Comment (garnier-quentin) [reply src/database/mongodb/mode/collectionstatistics.pm:208]: shards are optional.  So we don't want to have an error if we don't have it on a mongodb

Comment (lucie-dubrunfaut) [reply src/database/mongodb/mode/collectionstatistics.pm:208]: Indeed so maybe add filter condition to cach emptu result :

```
if (defined($self->{option_results}->{filter_shard}) && scalar(keys %{$self->{databases}->{$database}->{shards}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => 'No shard found with this filter');
        $self->{output}->option_exit();
    }
```

File: src/database/mongodb/mode/databasestatistics.pm [hunk 3]
```diff
@@ -125,25 +194,45 @@ sub manage_selection {
...
+                    views => $db_stats->{raw}->{$shard_name}->{views},
+                    documents => $db_stats->{raw}->{$shard_name}->{objects},
+                    storageSize => $db_stats->{raw}->{$shard_name}->{storageSize},
+                    indexSize => $db_stats->{raw}->{$shard_name}->{indexSize},
+                    dataSize => $db_stats->{raw}->{$shard_name}->{dataSize},
+                    indexes => $db_stats->{raw}->{$shard_name}->{indexes}
+                };
+            }
         }
     }
```
Comment (lucie-dubrunfaut) [inline src/database/mongodb/mode/databasestatistics.pm:238]: Same here if filter shard is asked an error message if filter result is empty :

```
    if (defined($self->{option_results}->{filter_shard}) && scalar(keys %{$self->{shards}}) <= 0) {
        $self->{output}->add_option_msg(short_msg => 'No shard found with this filter');
        $self->{output}->option_exit();
    }
```

**Review by lucie-dubrunfaut — APPROVED** [review]

**Pull request merged**